src/node_perf.cc

```diff
@@ -366,6 +366,24 @@ void Timerify(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(wrap);
 }
 
+// Notify a custom PerformanceEntry to observers
+void Notify(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Utf8Value type(env->isolate(), args[0]);
+  Local<Value> entry = args[1];
+  PerformanceEntryType entryType = ToPerformanceEntryTypeEnum(*type);
+  AliasedUint32Array& observers = env->performance_state()->observers;
+  if (entryType != NODE_PERFORMANCE_ENTRY_TYPE_INVALID &&
+      observers[entryType]) {
+    node::MakeCallback(env->isolate(),
+                       entry.As<Object>(),
+                       env->performance_entry_callback(),
+                       1, &entry,
+                       node::async_context{0, 0});
```

Comment:
Since this is called from the JS layer, I think using `env->performance_entry_callback()->Call(...)` would make more sense than calling `node::MakeCallback`.

And maybe leave a `// TODO` comment that, as a future optimization, maybe we could avoid the JS → C++ → JS boundary jumps here altogether?

Reply:
I'm not really sure, it might be related to the `async_context`. I believe the `PerformanceObserver` are also `AsyncRessource` so it might have a problem if we remove that ?

Reply:
@vmarchaud No, the async context is empty so it’s pointless. `MakeCallback` is only meaningful for bottom-level calls from C++ into JS, not when there’s already a JS stack below.

Reply:
I've switched to `env->performance_entry_callback()->Call(...)` but i'm really not familiar with C++ and the V8 API, could you tell me if it's good now ?